### PR TITLE
Backend for login and account page

### PR DESCRIPTION
**Pull request by mo-umar**
All of the backend part for the login and register pages.
Integrated backend with frontend login and register pages.
Fix CSS link in html file for login and register.

### CODE REVIEW

File: src/main/java/com/virtudoc/web/entity/UserAccount.java [hunk 1]
```diff
@@ -49,6 +74,28 @@ public UserAccount(String username, String password, String role) {
...
 
+    /**
+     * Constructor for UserAccount class. Utilizes the NewUserDTO to ensure that protected fields
+     * such as ROLE are not exposed in the controller.
+     * @param userDTO DTO object from frontend Model.
+     * @throws Exception Given password pair does not match.
+     * @see NewUserDTO
+     */
+    public UserAccount(NewUserDTO userDTO) throws Exception {
+        if (userDTO.getPassword().equals(userDTO.getConfirmedPassword())) {
```
Comment (ARMmaster17) [inline src/main/java/com/virtudoc/web/entity/UserAccount.java:85]: Sorry, this is a bug from my code. This conditional needs a `!` in front since we only want to throw an error if they don't match. In the interest of getting this PR out the door, this could be fixed in another PR later since it's not related to your new code.

File: src/main/java/com/virtudoc/web/entity/UserAccount.java [hunk 2]
```diff
@@ -30,6 +33,28 @@
...
+    @Column(unique = true)
+    @Email
+    @Size(max = 320) // Specified in RFC 2821 and RFC 3693.
+    private String email;
+
+    @NotNull
+    @NotEmpty
+    private String firstName;
+
+    @NotNull
```
Comment (ARMmaster17) [inline src/main/java/com/virtudoc/web/entity/UserAccount.java:47]: All the remaining test errors are from these new fields being invalid in `src/test/java/com/virtudoc/web/UserAccountTest.java` and `src/test/java/com/virtudoc/web/AuthenticationServiceTest.java`. It's up to you if you want to do constraint testing on all these new fields, but you will need to update the UserAccount objects in those tests to have non-null and non-empty values for each of these new fields. Once that's done the rest of those test errors should go away and the builds should come back green.

**Review by ARMmaster17 — APPROVED** [review]
Left some comments on approaches to fix the test errors. Otherwise LGTM
